lightly/models/modules/center.py

```diff
@@ -0,0 +1,94 @@
+from typing import Callable, Tuple
+
+import torch
+import torch.distributed as dist
+from torch import Tensor
+from torch.nn import Module
+
+
+class Center(Module):
+    """Center module to compute and store the center of a feature tensor as used
+    in DINO [0].
+
+    - [0]: DINO, 2021, https://arxiv.org/abs/2104.14294
+
+    Attributes:
+        size:
+            Size of the tracked center tensor. Dimensions across which the center
+            is computed must be set to 1. For example, if the feature tensor has shape
+            (batch_size, sequence_length, feature_dim) and the center should be computed
+            across the batch and sequence dimensions, the size should be
+            (1, 1, feature_dim).
+        mode:
+            Mode to compute the center. Currently only 'mean' is supported.
+        momentum:
+            Momentum term for the center calculation.
+        _register_buffer:
+            Deprecated, do not use. This argument is only kept for backwards
+            compatibility with DINOLoss.
+    """
+
+    def __init__(
+        self,
+        size: Tuple[int, ...],
+        mode: str = "mean",
+        momentum: float = 0.9,
+        _register_buffer: bool = True,
+    ) -> None:
+        super().__init__()
+
+        modes = ["mean"]
+        self._center_fn: Callable[[Tensor], Tensor]
+        if mode not in modes:
+            raise ValueError(f"Unknown mode '{mode}'. Valid modes are {modes}.")
+        if mode == "mean":
+            self._center_fn = self._center_mean
+        else:
+            raise NotImplementedError(f"Mode '{mode}' is not implemented.")
```

Comment:
Only the first error message will be raised. I would change this to one of these options:
```python
mode_to_center_fn = {
    "mean": self._center_mean,
}
if mode not in mode_to_center_fn:
    raise ValueError(f"Invalid mode '{mode}'. Valid modes are {list(mode_to_center_fn.keys())}.")
self._center_fn = mode_to_center_fn[mode]
```

or 
```
modes = ["mean"]
match mode:
    case "mean":
        self._center_fn = self._center_mean
    case _:
        raise ValueError(f"Invalid mode '{mode}'. Valid modes are {modes}.")
```

The first one has the advantage that you cannot accidentally forget to specify a new method at both places.